File: app/src/main/java/home/oleg/popularmovies/data/entities/MovieResponse.java

package home.oleg.popularmovies.data.entities;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class MovieResponse {

    private Integer page;
    private List<Result> results = null;
    private Integer totalResults;
    private Integer totalPages;

    public Integer getPage() {
        return page;
    }

    public void setPage(Integer page) {
        this.page = page;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    public Integer getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(Integer totalResults) {
        this.totalResults = totalResults;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public static class Result {

        @SerializedName(value = "poster_path")
        private String posterPath;

        @SerializedName(value = "adult")
        private Boolean adult;

        @SerializedName(value = "overview")
        private String overview;

        @SerializedName(value = "release_date")
        private String releaseDate;

        @SerializedName(value = "id")
        private Integer id;

        @SerializedName(value = "original_title")
        private String originalTitle;

        @SerializedName(value = "original_language")
        private String originalLanguage;

        @SerializedName(value = "title")
        private String title;

        @SerializedName(value = "backdrop_path")
        private String backdropPath;

        @SerializedName(value = "popularity")
        private Double popularity;

        @SerializedName(value = "vote_count")
        private Integer voteCount;

        @SerializedName(value = "video")
        private Boolean video;

        @SerializedName(value = "vote_average")
        private Double voteAverage;

        public String getPosterPath() {
            return posterPath;
        }

        public void setPosterPath(String posterPath) {
            this.posterPath = posterPath;
        }

        public Boolean getAdult() {
            return adult;
        }

        public void setAdult(Boolean adult) {
            this.adult = adult;
        }

        public String getOverview() {
            return overview;
        }

        public void setOverview(String overview) {
            this.overview = overview;
        }

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getOriginalTitle() {
            return originalTitle;
        }

        public void setOriginalTitle(String originalTitle) {
            this.originalTitle = originalTitle;
        }

        public String getOriginalLanguage() {
            return originalLanguage;
        }

        public void setOriginalLanguage(String originalLanguage) {
            this.originalLanguage = originalLanguage;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public String getBackdropPath() {
            return backdropPath;
        }

        public void setBackdropPath(String backdropPath) {
            this.backdropPath = backdropPath;
        }

        public Double getPopularity() {
            return popularity;
        }

        public void setPopularity(Double popularity) {
            this.popularity = popularity;
        }

        public Integer getVoteCount() {
            return voteCount;
        }

        public void setVoteCount(Integer voteCount) {
            this.voteCount = voteCount;
        }

        public Boolean getVideo() {
            return video;
        }

        public void setVideo(Boolean video) {
            this.video = video;
        }

        public Double getVoteAverage() {
            return voteAverage;
        }

        public void setVoteAverage(Double voteAverage) {
            this.voteAverage = voteAverage;
        }

        public String getReleaseDate() {
            return releaseDate;
        }

        public void setReleaseDate(String releaseDate) {
            this.releaseDate = releaseDate;
        }
    }
}

File: app/src/main/java/home/oleg/popularmovies/data/entities/MovieResponse.kt
package home.oleg.popularmovies.data.entities

import com.google.gson.annotations.SerializedName
import home.oleg.popularmovies.data.database.model.MovieDbModel
import home.oleg.popularmovies.domain.entities.Movie

data class MovieResponse(
        val page: Int? = null,
        val results: List<Result> = emptyList(),
        val totalResults: Int? = null,
        val totalPages: Int? = null) {

    data class Result(

            @SerializedName(value = "poster_path")
            val posterPath: String? = null,

            @SerializedName(value = "adult")
            val adult: Boolean? = null,

            @SerializedName(value = "overview")
            val overview: String? = null,

            @SerializedName(value = "release_date")
            val releaseDate: String? = null,

            @SerializedName(value = "id")
            val id: Long,

            @SerializedName(value = "original_title")
            val originalTitle: String? = null,

            @SerializedName(value = "original_language")
            val originalLanguage: String? = null,

            @SerializedName(value = "title")
            val title: String? = null,

            @SerializedName(value = "backdrop_path")
            val backdropPath: String? = null,

            @SerializedName(value = "popularity")
            val popularity: Double? = null,

            @SerializedName(value = "vote_count")
            val voteCount: Int? = null,

            @SerializedName(value = "video")
            val video: Boolean? = null,

            @SerializedName(value = "vote_average")
            val voteAverage: Double? = null
    )

}
